src/shared/components/buttons/primary.tsx

```diff
@@ -0,0 +1,40 @@
+import { StyleSheet, TouchableOpacity } from 'react-native';
+
+import { Typographies } from '../typographies';
+
+import { useStyles } from '~/shared/hooks/use-styles';
+import { createStylesheet } from '~/shared/styles/create-stylesheet';
+
+interface TextProps extends React.ComponentProps<typeof TouchableOpacity> {
```

Comment:
We should not extends all the properties of `TouchableOpacity`. Interface only the 3 properties we need. Then, remove the `{...props}` line 20 and pass only the `onPress` property.

_**same for other components**_